sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -127,6 +127,10 @@ def _from_message(msg):
     """
     # Convert ScalarMapContainer to dict.
     attributes = dict((key, msg.attributes[key]) for key in msg.attributes)
+    # Parse the PubSub message_id and add to attributes
+    if msg.message_id != None:
```

Comment:
Could message_id be None?

Reply:
It shouldn't be, was simply being defensive.  I've missed something off here however, and this is only working for the directrunner at the moment as I hadn't amended the _from_protobuf function.  The parsing of the inbound object is slightly different there, so I'm just checking whether this can be done in this manner now, and will amend (and remove the unnecessary check).

Reply:
Bit odd, the message_id is returned, but the value currently always seems to be blank, I've tried using the id_label parameter as well to specify message_id, but this still returns nothing.  Will continue to investigate.